src/utils/history.py

```
@@ -11,14 +13,56 @@
 
 logger = logging.getLogger(__name__)
 
+
+def parse_session_date_from_metadata(metadata: dict, fallback_to_created_at: bool = True) -> Optional[str]:
```

Pylint (3.3.7)

[convention] Line too long (107/100)

(C0301)